### PR TITLE
Vaibhav/milestone [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by VAIBHAVJINDAL3012**
# Description

Please provide a detailed description of what was done in this PR
Final PR for the milestone implementation. Milestone's all the functionality has been included in the Checkpoint module for the ease of Hardforking. This includes the following features.
1.Bridge-Milestone creation and broadcasting to tendermint
2.Handler and SideHandler/PostHandler for the milestone
3.Keeper module
4.Querier module

# Changes

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ✔] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Changes only for a subset of nodes

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Nodes audience

In case this PR includes changes that must be applied only to a subset of nodes, please specify how you handled it (e.g. by adding a flag with a default value...)

# Checklist

- [ ✔] I have added at least 2 reviewer or the whole pos-v1 team
- [ ✔] I have added sufficient documentation in code
- [ ✔] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply

# Cross repository changes

- [ ✔] This PR requires changes to bor 
  - In case link the PR here:
- [ ] This PR requires changes to matic-cli
  - In case link the PR here:

## Testing

- [ ✔] I have added unit tests
- [ ] I have added tests to CI
- [ ✔] I have tested this code manually on local environment
- [ ✔] I have tested this code manually on remote devnet using express-cli
- [ ] I have tested this code manually on mumbai
- [ ] I have created new e2e tests into express-cli

### Manual tests

Please complete this section with the steps you performed if you ran manual tests for this functionality, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: bridge/setu/processor/milestone.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+	return nil
+}
+
+// RegisterTasks - nil
+func (mp *MilestoneProcessor) RegisterTasks() {
+
+}
+
+// startPolling - polls heimdall and checks if new span needs to be proposed
```
Comment (manav2401) [inline bridge/setu/processor/milestone.go:53]: a minor typo: span -> milestone

File: bridge/setu/util/common.go [hunk 2]
```diff
@@ -38,10 +40,14 @@ const (
...
 	LastNoAckURL            = "/checkpoints/last-no-ack"
 	CheckpointParamsURL     = "/checkpoints/params"
+	MilestoneParamsURL      = "/milestone/params"
+	MilestoneCountURL       = "/milestone/count"
 	ChainManagerParamsURL   = "/chainmanager/params"
 	ProposersURL            = "/staking/proposer/%v"
+	MilestoneProposersURL   = "/staking/milestoneProposer/%v"
 	BufferedCheckpointURL   = "/checkpoints/buffer"
 	LatestCheckpointURL     = "/checkpoints/latest"
+	LatestMilestoneURL      = "/milestone/latest"
```
[the remark this answers is not in the document]
Comment (VAIBHAVJINDAL3012) [reply bridge/setu/util/common.go:50]: In Bridge, /milestone/{number} implementation is not required. We need to fetch the latest milestone only.

File: bridge/setu/processor/milestone.go [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+
+		if result.Count != 0 {
+			// fetch latest milestone
+			latestMilestone, err := util.GetLatestMilestone(mp.cliCtx)
+			if err != nil {
+				return err
+			}
+
+			if latestMilestone == nil {
+				return fmt.Errorf("Got nil result while fetching latest milestone")
```
Comment (JekaMas) [inline bridge/setu/processor/milestone.go:118]: errors.New or add additional data about the request

File: bridge/setu/processor/milestone.go [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+	if latestNum < startNum+milestoneLength+blocksConfirmation-1 {
+		return fmt.Errorf("Less than milestoneLength  Start=%v Latest Block=%v MilestoneLength=%v MaticChainConfirmation=%v", startNum, latestNum, milestoneLength, blocksConfirmation)
+	}
+
+	endNum := latestNum - blocksConfirmation
+
+	//fetch the endBlock+1 number instead of endBlock so that we can directly get the hash of endBlock using parent hash
+	block, err = mp.contractConnector.GetMaticChainBlock(big.NewInt(int64(endNum + 1)))
+	if err != nil {
+		return err
```
Comment (JekaMas) [inline bridge/setu/processor/milestone.go:160]: Shall we add context about the request? `("%d", endBlock+1)`

File: bridge/setu/processor/milestone.go [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+	}
+
+	endHash := block.ParentHash
+
+	milestoneId := uuid.NewRandom().String() + "-" + hmTypes.BytesToHeimdallAddress(endHash[:]).String()
```
Comment (JekaMas) [inline bridge/setu/processor/milestone.go:165]: Could you introduce a format and use rhen fmt.Sprintf(format, var1, var2...)?

**Review by JekaMas — APPROVED** [review]

**Review by temaniarpit27 — APPROVED** [review]

**Pull request merged**